netbox/netbox/settings.py

```diff
@@ -288,6 +293,30 @@ def _setting(name, default=None):
     'utilities.auth_backends.ViewExemptModelBackend',
 ]
 
+# GitHub repository for version check
+if GITHUB_REPOSITORY_API:
+    GITHUB_REPOSITORY_API = GITHUB_REPOSITORY_API.rstrip('/')
+    try:
+        scheme, netloc, path, query, fragment = urlsplit(GITHUB_REPOSITORY_API)
+    except ValueError:
+        raise ImproperlyConfigured("GITHUB_REPOSITORY_API must be a valid URL")
+
+    if scheme not in ('http', 'https'):
+        raise ImproperlyConfigured("GITHUB_REPOSITORY_API must be a valid http:// or https:// URL")
+
+    if not re.fullmatch(r'/repos/[A-Za-z0-9_.-]+/[A-Za-z0-9_.-]+', path):
```

Comment:
Might this be a little _too_ strict? I'm all for validation but I do worry that we might be shooting ourselves in the foot. For instance, does git/GitHub allow Unicode in repo names? I think if we keep this in place we should include a comment pointing to an authoritative spec for the naming format.

Reply:
I got this regex from GitHub docs somewhere, but I don't remember where… :/

Reply:
I have tested all characters that I could type on my keyboard (including a whole bunch of unicode) and every single character except those in the regex above are either replaced with `-` automatically or ignored by GitHub. I am therefore confident that the regex is accurate.